### ISSUE TITLE
Draft for new rule: Non-visual location reference alternative

### ISSUE THREAD

**Issue by ShadowBB** [act-rules/act-rules.github.io#714]
Draft for a new rule:

---
```
id: 825ule
name: Non-visual location reference alternative
rule_type: atomic

description: |
  This rule checks that when there is a references to a visual location of content, there are also non-visual indicators of the location.

accessibility_requirements:
  wcag20:1.3.3: # Sensory characteristics (A)
    forConformance: true
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed

  wcag-technique:G96: # Providing textual identification of items that otherwise rely only on sensory information to be understood.
    forConformance: true
    failed: not satisfied
    passed: further testing needed
    inapplicable: further testing needed

input_aspects:
  -  (DOM Tree, CSS Styling, Accessibility Tree, etc.,)

authors:
  -  Brian Bors
  -  Daniël strik
```
---

## Applicability

The rule applies to any reference to a visual location of content that is either visible or included in the accessibility tree.

## Expectation

Each target has a textual indicator that references the non-visual location of the content.

## Assumptions

_There are currently no assumptions_

## Accessibility Support

_There are no major accessibility support issues known for this rule._

## Background

- TODO We will add links as soon as the new template merges

#### Passed Example 1

The content in the column on the right is indicated with the word "right" (which is a location indicator based on visual perception) but also indicated by referencing the header "Fuzzyness of cats".

```html
<head>
<style>
.col-container {
  display: table;
}
.col {
  display: table-cell;
  padding: 16px;
}
</style>
</head>
<body>

<div class="col-container">
  <div class="col">
    <p>Read the content in the right column about the fuzzyness of cats.</p>
  </div>

  <div class="col">
    <h2>Fuzzyness of cats</h2>
    <p>Cats are usually fuzzy.</p>
  </div>
</div>

</body>
```

#### Passed Example 2

The content in the column on the right is indicated with the word "right" (which is a location indicator based on visual perception) but also indicated by referencing that the content can be found below this content in the DOM order.

```html
<head>
<style>
.col-container {
  display: table;
}
.col {
  display: table-cell;
  padding: 16px;
}
</style>
</head>
<body>

<div class="col-container">
  <div class="col">
    <p>Read the content in the right column (or below this paragraph) about the fuzzyness of cats</p>
  </div>

  <div class="col">
    <p>Cats are usually fuzzy.</p>
  </div>
</div>

</body>
```

#### Failed Example 1

The content in the column on the right is indicated with the word "right" (which is a location indicator based on visual perception) and there is no other indication of where to find the content.

```html
<head>
<style>
.col-container {
  display: table;
}
.col {
  display: table-cell;
  padding: 16px;
}
</style>
</head>
<body>

<div class="col-container">
  <div class="col">
    <p>Read the content in the right column about the fuzzyness of cats.</p>
  </div>

  <div class="col">
    <p>Cats are usually fuzzy.</p>
  </div>
</div>

</body>
```

#### Inapplicable Example 1

The content in the column on the right is not indicated with a location indicator based on visual perception.

```html
<head>
<style>
.col-container {
  display: table;
}
.col {
  display: table-cell;
  padding: 16px;
}
</style>
</head>
<body>

<div class="col-container">
  <div class="col">
    <p>Read the content below about the fuzzyness of cats.</p>
  </div>

  <div class="col">
    <p>Cats are usually fuzzy.</p>
  </div>
</div>

</body>
```

#### Inapplicable Example 2
The content in the column on the right is indicated with the word "right" (which is a location indicator based on visual perception) and there is no other indication of where to find the content. However the indicator is both visually hidden and not included in the accessibility tree.

```html
<head>
<style>
.col-container {
  display: table;
}
.col {
  display: table-cell;
  padding: 16px;
}
</style>
</head>
<body>

<div class="col-container">
  <div class="col">
    <p style="display:none;">Read the content in the right column about the fuzzyness of cats.</p>
  </div>

  <div class="col">
    <p>Cats are usually fuzzy.</p>
  </div>
</div>

</body>
```

**Comment by ShadowBB** [comment]
Comments from Wilco Fiers:

Hey Brian,
On principle, sure, this is a good rule to write. I think you're going to have a hard time defining "reference to a visual location" in an objective way. The ACT Rules Format requires that the applicability is objective, which it seems to me is going to be quite difficult in this case.

Wilco

**Comment by ShadowBB** [comment]
Comments from Anne Thyme Nørregaard:

Hi Brian,

We have discussed this proposal internally here at Siteimprove, and our immediate feedback is that the rule is interesting, but seems like it can get you into some dangerous places. One of our questions was how this rule can ever get unambiguous, when “right” is highlighted as a fail and “below” as a pass. And how do you define these things in a way that work across languages, and possibly for (semi-)automation?

Best Regards,

**Comment by WilcoFiers** [comment]
Alternative proposal

> Applicability
> 
> Any text node that includes one of the [visual reference words](), that is [visible]() or [included in the accessibility tree]()
> 
> Expectation
> 
> Each target that describes any [web content]() through the use of the [visual reference words](), is included in a paragraph that also describes that [web content]() by a non-visual property. Except if the target is not part of an instruction about [web content]().

**Comment by Jym77** [comment]
Closed by #832.
